src/pylutron_caseta/pairing.py

```diff
@@ -61,7 +61,7 @@ async def async_read_json(self, timeout):
 
     async def async_write_json(self, obj):
         """Write an object."""
-        buffer = f"{json.dumps(obj)}\r\n".encode("ASCII")
+        buffer = json.dumps(obj) + b"\r\n"
         self._writer.write(buffer)
```

Comment:
I think the payload is small here so it probably doesn't matter but if its larger than 2048 bytes `.writelines()` can be used instead as it does [zero copy writes](https://github.com/python/cpython/pull/31871). Otherwise `json.dumps(obj) + b"\r\n"` will result in `memcpy` to create a new immutable bytes string.

Reply:
`self._write.writelines((json.dumps(obj), b"\r\n"))` would be a tiny but faster since tuples are a bit cheaper than lists

Just but only if the average resulting bytes is > 2048 in length as otherwise it won't matter and you can ignore this suggestion.

Reply:
There are a couple cases in there that makes sense (aka the LeapProtocol.request hotpath, not so much the test cases 😆).